modules/host/contractmanager/persist_test.go

```diff
@@ -26,6 +26,365 @@ func (*dependencyNoRecheck) Disrupt(s string) bool {
 	return false
 }
 
+func Test_savedStorageFolder_Equals(t *testing.T) {
+	tests := []struct {
+		name     string
+		folder   savedStorageFolder
+		expected savedStorageFolder
+		want     bool
+	}{
+		{
+			name: "equal",
+			folder: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			expected: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			want: true,
+		},
+		{
+			name: "diff path",
+			folder: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			expected: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/02",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			want: false,
+		},
+		{
+			name: "diff index",
+			folder: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			expected: savedStorageFolder{
+				Index: 0,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			want: false,
+		},
+		{
+			name: "diff usage",
+			folder: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					1,
+					2,
+					3,
+					4,
+					5,
+				},
+			},
+			expected: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.folder.Equals(tt.expected); got != tt.want {
+				t.Errorf("savedStorageFolder.Equals() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_savedSettings_Equals(t *testing.T) {
+	tests := []struct {
+		name     string
+		settings savedSettings
+		expected savedSettings
+		want     bool
+	}{
+		{
+			name: "equal",
+			settings: savedSettings{
+				SectorSalt: crypto.HashBytes([]byte{1}),
+				StorageFolders: []savedStorageFolder{
+					{
+						Index: 1,
+						Path:  "/tmp/storage/01",
+						Usage: []uint64{
+							5,
+							1,
+							2,
+							3,
+							4,
```

Comment:
ok, now I'm just being picky, but it bothers me to have these on separate lines. I would probably condense this all the way down to:

```go
{Index: 1, Path:  "/tmp/storage/01", Usage: []uint64{5, 1, 2, 3, 4}},
```
this will make any differences between the two objects more readily apparent as well.